docs/pages/management/operations/self-hosted-automatic-agent-updates.mdx

```diff
@@ -0,0 +1,117 @@
+page](../../architecture/automatic-agent-update.mdx) describes how agent
+updating works.
+
+This guide covers how to set up the automatic update infrastructure. If this is
+already done, or you are a Teleport Cloud user, you can directly
+[enroll agents into automatic updates](./enroll-agent-into-automatic-updates.mdx).
+
+## Requirements
+
+- Self-hosted Teleport cluster running.
+- `tctl` admin rights.
+- Either:
+  - a public S3/GCS bucket,
+  - a web server accessible from all agents with valid TLS certificates.
+
+## Step 1/3. Create release channel files
+
+A release channel contains two pieces of information: the targeted version
+and if the update is critical. Updaters subscribe to a release channel and will
+update to the provided version during a maintenance window if possible. If the
+update is critical, updaters will ignore the maintenance schedule and update as
+soon as possible.
+
+Create a directory for the new release channel `current`.
+
+```code
+$ mkdir current/
+```
+
+Make the `current` release channel target the version (=teleport.version=):
+
+```code
+$ echo -n "(=teleport.version=)" > current/version
+```
+
+And mark the update as not critical:
+
+```code
+$ echo -n "no" > current/critical
+```
+
+## Step 2/3. Host the release channel files on the version server
+
+The release channel must be hosted on a webserver with trusted TLS certificates
+and reachable by all agents.
+
+Public cloud buckets like Amazon S3 or Google Cloud Storage are good
+candidates as they provide reliable low-maintenance file hosting.
+
+You can also serve the files with a regular webserver like `nginx`.
+The release channel needs to be served over HTTPS.
+
+Choose a way to serve the release channel and setup the hosting.
+
+The webserver must answer the following queries:
+
+```code
+$ curl https://<hosting-domain-and-path>/current/version
+(=teleport.version=)
+
+$ curl https://<hosting-domain-and-path>/current/critical
+no
+```
+
+The web server serving the release channel is called the version server.
+Save the version server domain and release channel name (here respectively
+`<hosting-domain-and-path>` and `current`) as they will be required
+later to configure the agent updaters.
+
+## Step 3/3. Configure the maintenance schedule
+
+At this point the updaters can be configured to pull the version from the
+release channel and update the agents. However, they still don't know when
+they should perform updates.
+
+Agents can retrieve the maintenance schedule from the Teleport cluster and
+pass it to the updater. In this step you'll configure the maintenance
+schedule for the whole cluster.
+
+Create the following `cmc.yaml` manifest allowing maintenances on Monday, Wednesday
+and Friday between 02:00 and 03:00 UTC.
+
+(!docs/pages/includes/cluster-maintenance-config-spec.mdx!)
+
+Finally, apply the manifest using `tctl`:
+
+```code
+$ tctl create cmc.yaml
+maintenance window has been updated
+```
+
+## Next steps
```

Comment:
I dont' see a section on troubleshooting.  What if the automatic upgrade fails.  What logs can you look at. how do you recover.

Reply:
added a few lines on how to trigger an update check manually and check the logs